server/src/main/java/org/elasticsearch/indices/TimestampFieldMapperService.java

```diff
@@ -172,8 +174,12 @@ public DateFieldMapper.DateFieldType getTimestampFieldType(Index index) {
         if (future == null || future.isDone() == false) {
             return null;
         }
-        // call non-blocking actionResult() as we could be on a network or scheduler thread which we must not block
-        return future.actionResult();
+        // call non-blocking result() as we could be on a network or scheduler thread which we must not block
+        try {
+            return future.result();
+        } catch (ExecutionException e) {
+            throw new UncategorizedExecutionException("An error occurred fetching timestamp field type for " + index, e);
+        }
```

Comment:
This is not quite the same as before, I believe previously, if there was a `RuntimeException` it would be thrown directly, and if there was a checked exception the behaviour would be consistent with the above. But looking up the call stack there doesn't appear to be any behaviour specific to the exception thrown. I'm happy to try and restore the original behaviour but fear it might just be adding noise to be consistent with something that nobody was depending on.

Reply:
Yes I think this is ok, indeed possibly preferable, because now we get the call stack to the `.result()` call as well as the one which failed the future.